### ISSUE TITLE
Weird behaviour with unsaved tabs

### ISSUE THREAD

**Issue by smbea** [camunda/camunda-modeler#2654]
__Describe the Bug__

Saved tabs appear as unsaved when opening and closing new diagrams.

__Steps to Reproduce__

https://user-images.githubusercontent.com/25825387/147941202-001efccc-c106-4dd3-9729-adba04ed84af.mov

1. open a new diagram
2. make a change and save file
3. open a new diagram and close it
4. see that first tab saved appears as unsaved

__Expected Behavior__

The tab saved shouldn't be displayed as unsaved - shouldn't display blue marker

__Environment__

- OS: MacOS 11.6
- Camunda Modeler Version: latest on develop

**Comment by andreasgeier** [comment]
I can confirm this bug. The dirty marker also appears on newly created editor tabs (that were never edited) when saving a file in another tab.

**Comment by andreasgeier** [comment]
I suggest this bug as a spring cleaning topic.

**Comment by smbea** [comment]
I believe the previously saved tab is marked as dirty in `showTab`, after the state is updated -> https://github.com/camunda/camunda-modeler/blob/28cec1fa11e79cd1e47cad6bb3660fd0a6b10b7d/client/src/app/App.js#L352

**Comment by smbea** [comment]
I believe the previously saved tab is marked as dirty in `showTab`, after the state is updated -> https://github.com/camunda/camunda-modeler/blob/28cec1fa11e79cd1e47cad6bb3660fd0a6b10b7d/client/src/app/App.js#L352

I couldn't figure out why it is causing this, though.

**Comment by marstamm** [comment]
This is a tricky one :smile: 

Here is what's causing it:

To figure out if changes were made, we are using the stack Index:
https://github.com/camunda/camunda-modeler/blob/9cf95538d592946399cdbcd98c174ddc119e9a25/client/src/app/tabs/bpmn/BpmnEditor.js#L478-L487

We cache it when we safe the XML:
https://github.com/camunda/camunda-modeler/blob/9cf95538d592946399cdbcd98c174ddc119e9a25/client/src/app/tabs/bpmn/BpmnEditor.js#L566-L577

However, as you can see in the linked file, the saved stackId is the one **before** we export the XML. During export, some things like `elements.alignToOrigin` are executed on the stack, so the cached ID is not correct anymore.

Basically, the Editor always things its dirty after saving, but the Tab update is only triggered if something in the Editor happens, like an `attach` event when switching tabs.

**Comment by marstamm** [comment]
I'll create a PR for a fix
